src/services/organizeImports.ts

```diff
@@ -927,3 +829,114 @@ function getTopLevelExportGroups(sourceFile: SourceFile) {
     }
     return flatMap(topLevelExportGroups, exportGroupDecls => groupByNewlineContiguous(sourceFile, exportGroupDecls));
 }
+
+/** @internal */
+export function detectModuleSpecifierCaseBySort(importDeclsByGroup: (readonly AnyImportOrRequireStatement[])[], comparersToTest: Comparer<string>[]) {
+    const moduleSpecifiersByGroup: string[][] = [];
+    importDeclsByGroup.forEach(importGroup => {
+        // Turns importDeclsByGroup into string[][] of module specifiers by group to detect sorting on module specifiers
+        moduleSpecifiersByGroup.push(getModuleNamesFromDecls(importGroup));
+    });
+    return detectCaseSensitivityBySort(moduleSpecifiersByGroup, comparersToTest);
+}
+
+/** @internal */
+export type TypeOrder = "first" | "last" | "inline";

```

Comment:
I would think that this is actually a part of the services types / protocol somewhere; should this be moved?

Reply:
Actually, I'm not seeing any changes which affect the protocol or service types, which I expected given we're adding a new preference (like Ron's past PR touching this).

Reply:
No, I'm wrong; the ordering was added in a previous commit. But this duplicate declaration does feel suspect.

Reply:
It does sound better that way. I hadn't named the type in the previous commit